README.md

```diff
@@ -0,0 +1,125 @@
+# JSInterop TS defs
+
+This annotation processor generates TypeScript definitions from Java classes annotated with JsInterop annotations.
+
+### Maven dependencies
+
+- Annotations :
+
+```xml
+
+<parent>
+    <artifactId>tsinterop-annotations</artifactId>
+    <groupId>com.vertispan.tsdefs</groupId>
+    <version>HEAD-SNAPSHOT</version>
+</parent>
+```
+- Processor
+
+```xml
+
+<depndency>
+    <artifactId>tsinterop-processor</artifactId>
+    <groupId>com.vertispan.tsdefs</groupId>
+    <version>HEAD-SNAPSHOT</version>
+</depndency>
+```
+
+### Controlling the output
+
+The output from this processor can be controlled by adding custom annotations to the java classes.
+
+- `@TsModule` : use this annotation on a package-info to define the name of the generated `.d.ts` file, by default the name is `types`.
+- `@TsName` : Use this annotation on a java type to change the its name or name space in the typescript definitions.
+
+    **Example** :
+    
+    The following Java type
+
+    ```java
+    import com.vertispan.tsdefs.annotations.TsName;
+    import jsinterop.annotations.JsPackage;
+    import jsinterop.annotations.JsType;
+    
+    @JsType(name = "SimpleType", namespace = "com.vertispan")
+    @TsName(name = "TypeSimple", namespace = JsPackage.GLOBAL)
+    public class SimpleJsType {
+    
+        public String name;
+    }
+    ```
+    
+    Will be exported as TypeScript definition as :
+    
+    ```typescript
+    export class TypeSimple {
+        name:string;
+    
+        constructor();
+    }
+    ```
+- `@TsTypeRef` : Use this annotation on a type, parameter, method, or field to override the element type by the type of another element.
+
+    **Example** :
```

Comment:
I'm not sure about these examples - do we think we often want to emit TS to global when the JS is in a namespace?

Reply:
The idea of the example is to show how the annotation works not to describe a real case.